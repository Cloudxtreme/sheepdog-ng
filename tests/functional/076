#!/bin/bash

# Test sheep recovery logic for erasure coding

. ./common

for i in `seq 0 45`; do
    _start_sheep $i
done

_wait_for_sheep 46

_cluster_format -c 16:15

$DOG vdi create test 40M
_vdi_list
_random | $DOG vdi write test
$DOG vdi read test | md5sum > $STORE/csum1

for i in `seq 1 15`;do _kill_sheep $i;done
$DOG vdi read test | md5sum > $STORE/csum2

diff -u $STORE/csum1 $STORE/csum2

_cleanup

for i in `seq 0 2`; do
    _start_sheep $i
done

_wait_for_sheep 3

_cluster_format -c 2:1

$DOG vdi create test 40M
_vdi_list
_random | $DOG vdi write test
$DOG vdi read test | md5sum > $STORE/csum1

_kill_sheep 1;
$DOG vdi read test | md5sum > $STORE/csum2

diff -u $STORE/csum1 $STORE/csum2
