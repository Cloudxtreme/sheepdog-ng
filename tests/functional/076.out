QA output created by 076
using backend plain store
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
  test         0   40 MB  0.0 MB  0.0 MB DATE   7c2b25  16:15              
using backend plain store
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
  test         0   40 MB  0.0 MB  0.0 MB DATE   7c2b25    2:1              
